docs/sphinx/user_guide/feature/view.rst

```diff
@@ -330,6 +330,60 @@ indices by :math:`N`, ::
     }
   }
 
+Index Layout
+^^^^^^^^^^^^
+
+``RAJA::IndexLayout`` is a layout that uses an indexing strategy for each
+dimension to determine how to map a given input index to an entry within a view.
+
+Three indexing strategies are natively supported in RAJA:
+DirectIndex, IndexList, and ConditionalIndexList. DirectIndex maps an input
+index to itself, and does not take any arguments in its constructor.  The
+IndexList strategy takes a reference to an array containing the list of relevant
+indices to map.  With this strategy, a given input index is mapped to the entry
+in its list corresponding to that index.  Lastly, the ConditionalIndexStrategy
+takes a reference to an array containing an index list.  When the referenced
+index list is non-null, the ConditionalIndex strategy is equivalent to that
+of the IndexList.  Otherwise, if the index list provided to the constructor is a
+null pointer, the ConditionalIndexList maps an input index to itself (same as
+the DirectIndex strategy).  The ConditionalIndexList strategy is useful when the
+index list is not initialized for some situations.
+
+A simple illustrative example is shown below::
+
+  int data[2][3];
+
+  for (int i = 0; i < 2; i ++ ) {
+    for (int j = 0; j < 3; j ++ ) {
+      // fill data[i][j]...
+    }
+  }
+
+  int index_list[2] = {1,2};
+
+  auto index_tuple = RAJA::tuple<RAJA::DirectIndex<>, RAJA::IndexList<>>(
+                      RAJA::DirectIndex<>(), RAJA::IndexList<>{&index_list[0]});
+			   
+  auto index_layout = RAJA::make_index_layout(index_tuple, 2, 3);
+  auto view = RAJA::make_index_view(&data[0][0], index_layout);
+
+  int val0 = view(1,0); // val0 is entry in data[1][1]
+  int val1 = view(1,1); // val1 is entry in data[1][2]

```

Comment:
I'm not sure if its more clear with asserts or not but it's a way to show equality in a code example.
```suggestion
  assert( view(1,0) == data[1][1] );
  assert( &view(1,1) == &data[1][2] );
```